pages/index.tsx

```diff
@@ -52,33 +52,42 @@ export default function Index(props: Props): JSX.Element {
   )
 }
 
-export async function getStaticProps() {
+export async function getStaticProps(): Promise<any> {
   const date = new Date().toISOString()
-  const eventUrl = `https://api.eventyay.com/v1/events?cache=true
-  &filter=[{"and":[{"name":"state","op":"eq","val":"published"},{"name":"privacy","op":"eq","val":"public"},{"name":"is-featured","op":"eq","val":true}]},{"or":[{"name":"ends-at","op":"ge","val":"${date}"}]}]
-  &page[size]=6
-  &public=true
-  &sort=starts-at`
-  const upcomingEventUrl = `https://api.eventyay.com/v1/events/upcoming?cache=true&page[size]=3&public=true&upcoming=true`
-  const groupsUrl = `https://api.eventyay.com/v1/groups?cache=true&filter=[{"name":"is-promoted","op":"eq","val":true}]&include=user,follower&page[size]=3&public=true`
 
-  const [event, eventErr] = await fetcher(eventUrl)
+  const [event, eventErr] = await fetcher(
+    'events',
+    'filter=%5B%7B%22and%22%3A%5B%7B%22name%22%3A%22state%22%2C%22op%22%3A%22eq%22%2C%22val%22%3A%22published%22%7D%2C%7B%22name%22%3A%22privacy%22%2C%22op%22%3A%22eq%22%2C%22val%22%3A%22public%22%7D%2C%7B%22or%22%3A%5B%7B%22name%22%3A%22is-featured%22%2C%22op%22%3A%22eq%22%2C%22val%22%3Atrue%7D%2C%7B%22name%22%3A%22is-promoted%22%2C%22op%22%3A%22eq%22%2C%22val%22%3Atrue%7D%2C%7B%22and%22%3A%5B%7B%22name%22%3A%22logo-url%22%2C%22op%22%3A%22ne%22%2C%22val%22%3Anull%7D%2C%7B%22name%22%3A%22original-image-url%22%2C%22op%22%3A%22ne%22%2C%22val%22%3Anull%7D%2C%7B%22name%22%3A%22event-topic%22%2C%22op%22%3A%22ne%22%2C%22val%22%3Anull%7D%2C%7B%22name%22%3A%22event-sub-topic%22%2C%22op%22%3A%22ne%22%2C%22val%22%3Anull%7D%2C%7B%22name%22%3A%22event-type%22%2C%22op%22%3A%22ne%22%2C%22val%22%3Anull%7D%5D%7D%5D%7D%2C%7B%22name%22%3A%22is-sessions-speakers-enabled%22%2C%22op%22%3A%22eq%22%2C%22val%22%3Atrue%7D%2C%7B%22name%22%3A%22is-demoted%22%2C%22op%22%3A%22eq%22%2C%22val%22%3Afalse%7D%2C%7B%22name%22%3A%22speakers-call%22%2C%22op%22%3A%22has%22%2C%22val%22%3A%7B%22name%22%3A%22privacy%22%2C%22op%22%3A%22eq%22%2C%22val%22%3A%22public%22%7D%7D%2C%7B%22name%22%3A%22speakers-call%22%2C%22op%22%3A%22has%22%2C%22val%22%3A%7B%22name%22%3A%22starts-at%22%2C%22op%22%3A%22le%22%2C%22val%22%3A%222022-03-09T11%3A52%3A46.823Z%22%7D%7D%2C%7B%22name%22%3A%22speakers-call%22%2C%22op%22%3A%22has%22%2C%22val%22%3A%7B%22name%22%3A%22ends-at%22%2C%22op%22%3A%22ge%22%2C%22val%22%3A%222022-03-09T11%3A52%3A46.823Z%22%7D%7D%2C%7B%22name%22%3A%22speakers-call%22%2C%22op%22%3A%22has%22%2C%22val%22%3A%7B%22name%22%3A%22announcement%22%2C%22op%22%3A%22ne%22%2C%22val%22%3Anull%7D%7D%5D%7D%2C%7B%22or%22%3A%5B%7B%22name%22%3A%22ends-at%22%2C%22op%22%3A%22ge%22%2C%22val%22%3A%222022-03-09T11%3A52%3A46.822Z%22%7D%2C%7B%22name%22%3A%22starts-at%22%2C%22op%22%3A%22ge%22%2C%22val%22%3A%222022-03-09T11%3A52%3A46.822Z%22%7D%5D%7D%5D&include=event-topic%2Cevent-sub-topic%2Cevent-type%2Cspeakers-call&page%5Bsize%5D=6&public=true&sort=starts-at'
```

Comment:
why??

Reply:
I just added it for testing. We should refactor it in the way I did below for groups with correct parameters.